### PR TITLE
BF: Make gitrepo behave with special remotes (that all have no refs)

### PR DESCRIPTION
**Pull request by hanke**

### CODE REVIEW

File: datalad/support/gitrepo.py [hunk 1]
```diff
@@ -526,8 +526,15 @@ def git_get_remote_branches(self):
...
-                remote_branches.append(ref.name)
+            try:
+                for ref in remote.refs:
+                    remote_branches.append(ref.name)
+            except AssertionError as e:
+                if e.message.endswith("did not have any references"):
+                    # this will happen with git annex special remotes
+                    pass
+                else:
+                    raise e
```
Comment (yarikoptic) [inline datalad/support/gitrepo.py]: and it is all kosher if 1 is annex special and others regular?

if only there was a test which should be real easy to create

Comment (hanke) [inline datalad/support/gitrepo.py]: This try block is per remote -- either a remote is special or not. Manual test had both types of remotes. My test repo creation foo is not yet advanced enough to produce such a test. I'll keep praying for enlightenment... ;-)

Comment (yarikoptic) [inline datalad/support/gitrepo.py]: ok.  I will check your knees when I come!

**Comment by hanke** [comment]
Thanks!

**Pull request merged**